src/channels/channel.ts

```diff
@@ -136,4 +164,32 @@ export class Channel {
             Log.info(`[${new Date().toLocaleTimeString()}] - ${socket.id} joined channel: ${channel}`);
         }
     }
+
+    /**
+     * Check if client is a client event
+     *
+     * @param  {string} event
+     * @return {boolean}
+     */
+    isClientEvent(event: string): boolean {
+        let isClientEvent = false;
+
+        this._clientEvents.forEach(clientEvent => {
+            let regex = new RegExp(clientEvent.replace('\*', '.*'));
+            if (regex.test(event)) isClientEvent = true;
+        });
+
+        return isClientEvent;
+    }
+
+    /**
+     * Check if a socket has joined a channel.
+     *
+     * @param socket
+     * @param channel
+     * @returns {boolean}
+     */
+    isInChannel(socket: any, channel: string): boolean {
+        return this.io.sockets.adapter.sids[socket.id][channel];
+    }
```

Comment:
Have you been able to test that out? Socket.io code is extra complex and that was the first thing that I could find that proved to work.